nasa/cmr-stac
# CMR-10132: Fix stac-browser Stac item Search unmapped CMR keys

# Overview

### What is the feature?

Update sorting maps to include a `datetime` mapping to CMR's `startDate` field. Refactor sortKeys to map to CMR fields depending if the search is for `items` or for `collections` so that we can sort by `id` field accordingly

### What is the Solution?

Adding search context to the query param and a util to direct the request to the correct CMR key field. Also adding a mapping for the `datetime` field

### What areas of the application does this impact?

# Testing

### Reproduction steps

- **Environment for testing:**
- **Collection to test with:**

1. Navigate to https://radiantearth.github.io/stac-browser/#/external/http:/localhost:3000/stac/ARCTEST ensure that using the `Id` for the collection search sorting works
2. ensure that using the `Id` for the items search sorting works
3. Ensure that sorting for the collection search sorting  by Date and time works as expected
4. Ensure that  sorting for the item search sorting  by Date and time works as expected

### Attachments

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.

# Checklist
